src/Graph.ts

```diff
@@ -453,7 +446,22 @@ export default class Graph {
 
 			module.dependencies.forEach(depModule => {
 				if (!depModule.isExternal) {
-					visit(<Module>depModule, seen);
+					if (depModule.id in parents) {
+						if (!allSeen[depModule.id]) {
+							let path = [depModule.id];
+							let id = module.id;
+							while (id !== depModule.id) {
+								path.push(id);
+								id = parents[id];
+								if (!id) break;
```

Comment:
This should never happen imho.

Reply:
When the top-level import is hit I think?

Reply:
Yeah, which means there is no cycle, in which case I think this code is not triggered.

Reply:
Right of course. How's the output looking?

Reply:
For example like this:

> (!) Circular dependency: node_modules/readable-stream/lib/_stream_readable.js -> node_modules/readable-stream/lib/_stream_duplex.js -> commonjs-proxy:/tmp/rollup-stream-issue-repro/node_modules/readable-stream/lib/_stream_readable.js -> node_modules/readable-stream/lib/_stream_readable.js

Reply:
Nice, if we can just get a test with one of these outputs sounds like we're nearly there?

Reply:
I just updated the function tests to expect this warning, so I think that's covered.

Reply:
Great I guess we can remove this break then as well?

Reply:
If my assumption is wrong this will end in an infinite loop, so I would rather like to keep it. Instead of breaking we could just return, though, and maybe emit a warning, since hitting this is a clear sign of a mistake in my reasoning.

Reply:
I'm still wondering about the case where a imports b and c, and b imports c. As our visit function itself is written as a pre-order visitor, we will call:

* visit(a) for the top-level
* visit(b) for a
* visit(c) for b
* then visit(c) for a again

So I still seem to think that we'd have `c` populated in parents by the time the `visit(c)` for `a` happens. Which then would trigger this case?

Reply:
No, `visit(c)` for `b` would have finished by then, so that `c` would be in `allSeen`.

Reply:
Ahh right. Interesting how allSeen gives us some great higher level properties, which was completely unintended :P